src/lib/vocab/assignment/assignmentServices.ts

```diff
@@ -15,33 +18,16 @@ export const CreateAssignmentService: ServiceByName['CreateAssignment'] = {
 	event: CreateAssignment,
 	perform: ({transact, params}) =>
 		transact(async (repos) => {
-			const {community_id, persona_id, role_id} = params;
-			log.trace('[CreateAssignment] creating assignment', persona_id, community_id);
-
-			// Personal communities disallow assignments as a hard rule.
+			const {actor, community_id, persona_id, role_id} = params;
+			log.trace('[CreateAssignment] creating assignment for persona & role', persona_id, role_id);
+			log.trace('[CreateAssignment] checking policy', actor, community_id);
 			const community = unwrap(await repos.community.findById(community_id));
 			if (!community) {
 				return {ok: false, status: 400, message: 'community not found'};
```

Comment:
```suggestion
				return {ok: false, status: 404, message: 'community not found'};
```

checked elsewhere, we do 404s for this in a 5 other places

Reply:
As a wise person once said, "if you put it off long enough, someone smarter might do it first" :D

I agree though about probably wanting to do it sooner rather than later. I know we have a bunch of spots in the UX where error messages don't even propagate to users, so it might make sense to do some of that planning/refactoring at the same time.